tests/lib/rules/no-object-constructor.js

```diff
@@ -104,6 +104,260 @@ ruleTester.run("no-object-constructor", rule, {
                     output: "({} instanceof Object);"
                 }]
             }]
-        }
+        },
+
+        ...[
+
+            // Semicolon required before `({})` to compensate for ASI
+            {
+                code: `
+                foo
+                Object()
+                `
+            },
+            {
+                code: `
+                foo()
+                Object()
+                `
+            },
+            {
+                code: `
+                new foo
+                Object()
+                `
+            },
+            {
+                code: `
+                (a++)
+                Object()
+                `
+            },
+            {
+                code: `
+                ++a
+                Object()
+                `
+            },
+            {
+                code: `
+                const foo = function() {}
+                Object()
+                `
+            },
+            {
+                code: `
+                const foo = class {}
+                Object()
+                `
+            },
+            {
+                code: `
+                foo = this.return
+                Object()
+                `
+            },
+            {
+                code: `
+                var yield = bar.yield
+                Object()
+                `
+            },
+            {
+                code: `
+                var foo = { bar: baz }
+                Object()
+                `
+            }
+        ].map(props => ({
+            ...props,
+            errors: [{
+                messageId: "preferLiteral",
+                suggestions: [{
+                    desc: "Replace with '({})', add preceding semicolon.",
+                    messageId: "useLiteralAfterSemicolon"
+                }]
```

Comment:
Suggestion test objects should generally have the `output` property (and it will become mandatory after we implement https://github.com/eslint/rfcs/pull/84).